src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -258,7 +258,7 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             var typeParameters = localFunction.TypeParameters;
             if (typeParameters.Any(typeParameter => typeParameter.HasUnmanagedTypeConstraint))
             {
-                _factory.CompilationState.ModuleBuilderOpt?.EnsureIsUnmanagedAttributeExists();
+                _factory.CompilationState.ModuleBuilderOpt?.EnsureIsUnmanagedAttributeExists(recordUsage: true);
```

Comment:
>EnsureIsUnmanagedAttributeExists(recordUsage: true) [](start = 60, length = 51)

do we ever pass `false` to this API?
(same question for `EnsureNullableAttributeExists` and `EnsureIsReadOnlyAttributeExists` below)

Reply:
> do we ever pass false to this API?

Not at the moment. However, since the API is defined in a different component, which participates in phases beyond lowering, I think there is a value in making consumers of the API aware of its side-effect and providing an option to opt-out.  

---
In reply to: [354490448](https://github.com/dotnet/roslyn/pull/40138#discussion_r354490448) [](ancestors = 354490448)